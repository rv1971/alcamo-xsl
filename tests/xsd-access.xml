<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc
    xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#"
    xmlns:axsd="tag:rv1971@web.de,2021:alcamo-xsl:xsd#"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xmlns:axsd="tag:rv1971@web.de,2021:alcamo-xsl:xsd#"
      xmlns:xsd="http://www.w3.org/2001/XMLSchema"
      xml:id="xslt"
      version="1.0"
      exclude-result-prefixes="axsd xsd">
    <xsl:import href="../xsl/xsd-access.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
     </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="@key"/>
        <xsl:text> </xsl:text>
        <xsl:value-of select="@value"/>
        <xsl:text> </xsl:text>
        <xsl:value-of select="key(@key, @value)/@a:title"/>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <xsd:attribute name="foo" a:title="Foo"/>

  <xsd:attributeGroup name="bar" a:title="Bar" id="corge"/>

  <xsd:element name="foo" a:title="Foo element">
    <xsd:annotation id="CORGE" a:title="Corge annotation">
      <xsd:documentation>Lorem ipsum</xsd:documentation>
    </xsd:annotation>
  </xsd:element>

  <xsd:group name="qux" a:title="Qux"/>

  <xsd:simpleType name="quux" a:title="Quux"/>

  <xsd:complexType name="baz" a:title="Baz"/>

  <a:data>
    <a:item key="axsd:attributes" value="foo"/>
    <a:item key="axsd:attributeGroups" value="bar"/>
    <a:item key="axsd:elements" value="foo"/>
    <a:item key="axsd:groups" value="qux"/>
    <a:item key="axsd:types" value="quux"/>
    <a:item key="axsd:types" value="baz"/>
    <a:item key="axsd:types" value="corge"/>
    <a:item key="axsd:elements-with-id" value="corge"/>
    <a:item key="axsd:elements-with-id" value="CORGE"/>
  </a:data>
</a:doc>
