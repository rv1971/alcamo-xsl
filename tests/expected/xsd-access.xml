<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>axsd:attributes foo Foo</a:item>
  <a:item>axsd:attributeGroups bar Bar</a:item>
  <a:item>axsd:elements foo Foo element</a:item>
  <a:item>axsd:groups qux Qux</a:item>
  <a:item>axsd:types quux Quux</a:item>
  <a:item>axsd:types baz Baz</a:item>
  <a:item>axsd:types corge </a:item>
  <a:item>axsd:elements-with-id corge Bar</a:item>
  <a:item>axsd:elements-with-id CORGE Corge annotation</a:item>
</a:result>
